src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionsController.kt

```diff
@@ -35,40 +42,67 @@ class IdentificationSessionsController(
     }
 
     @PostMapping
-    fun createSession(@RequestBody createIdentitySessionRequest: CreateIdentitySessionRequest): Mono<ResponseEntity<CreateIdentitySessionResponse>> {
-        // Currently, returns a mock session response
-        return tcTokenUrlService.getTcTokenUrl()
-            .flatMap { tcTokenUrl ->
-                identificationSessionService.save(
-                    tcTokenUrl,
-                    createIdentitySessionRequest.refreshAddress,
-                    createIdentitySessionRequest.requestAttributes
-                )
-            }.map {
+    fun createSession(
+        @RequestBody createIdentitySessionRequest: CreateIdentitySessionRequest,
+        serverHttpRequest: ServerHttpRequest
+    ): Mono<ResponseEntity<CreateIdentitySessionResponse>> {
+        return identificationSessionService
+            .save(
+                refreshAddress = "https://localhost:8443/", // Currently a mock value, later will inject from env var mapped by API Key
+                requestAttributes = createIdentitySessionRequest.requestAttributes
+            ) // PublishOn is suggested by IntelliJ, maybe there is another way to resolve blocking URLEncoder.encode()
+            .publishOn(Schedulers.boundedElastic())
+            .map {
+                val encodedTcTokenUrl = URLEncoder.encode("${serverHttpRequest.uri}/$TCTOKEN_ENDPOINT/${it.useIDSessionId}", Utils.ENCODING)
```

Comment:
Is this URL working? From the code, I'd assume, that it isn't. It's missing `/api/v1/identification/sessions`.

Reply:
yes, this URL is working, my assumption since client requests to the endpoint `/api/v1/identification/sessions`, the `serverHttpRequest.uri` contains the correct path already.

<img width="916" alt="image" src="https://user-images.githubusercontent.com/9930966/184348918-5ba24172-58d0-4699-aaa5-55c4b55126e3.png">

Reply:
Wow, that's.. unexpected. And a bit error-prone, since we might change the route of this endpoint, so that it's no prefix of the `tcTokenUrl` anymore? Hmm... not sure about it here, but this feels too "magic" for me for being self-explanatory.

Reply:
I agree that the code is not really straight forward and we have to guess whether the `uri` contains the path or not, we can make it more explicit though.. since I am lack knowledge of Spring API, we will find a way

Reply:
Yes, let's make it explicit please. I guess this would also break if we hide our service behind a proxy (or something like that) and would return internal domain names.